storage/innobase/ut/ut0crc32.cc

```diff
@@ -733,27 +730,17 @@ ut_crc32_init()
 	probably kill your program.
 
 	*/
-#ifndef UNIV_DEBUG_VALGRIND
-	ut_crc32_sse2_enabled = (features_ecx >> 20) & 1;
-#endif /* UNIV_DEBUG_VALGRIND */
 
-	if (ut_crc32_sse2_enabled) {
+	if ((features_ecx >> 20) & 1) {
 		ut_crc32 = ut_crc32_hw;
 		ut_crc32_legacy_big_endian = ut_crc32_legacy_big_endian_hw;
 		ut_crc32_byte_by_byte = ut_crc32_byte_by_byte_hw;
+		ut_crc32_implementation = "Using SSE2 crc32 instructions";
 	}
 
-#endif /* defined(__GNUC__) && defined(__x86_64__) */
-
-#ifdef HAVE_CRC32_VPMSUM
-	ut_crc32_power8_enabled = true;
+#elif defined(HAVE_CRC32_VPMSUM)
```

Comment:
Above on x86, we are checking the availability of the hardware-assisted CRC32-C implementation with the CPUID instruction, while here for POWER, we are checking it already at compilation time.
I wonder if this would cause trouble for those who use precompiled binaries (such as those included in GNU/Linux distributions) on POWER. (If the build host does not have the CRC32 instructions, the slow implementation would always be used. If the build host has the instructions but the target system does not, it would fail.)
I leave this up to you, because I do am not that familiar with POWER. Maybe it could be addressed in a follow-up pull request?